iceoryx_posh/source/version/version_info.cpp

```diff
@@ -0,0 +1,120 @@
+    , m_versionMinor(versionMinor)
+    , m_versionPatch(versionPatch)
+    , m_versionTweak(versionTweak)
+    , m_buildDateString(buildDateString)
+    , m_commitIdString(commitIdString)
+{
+}
+
+VersionInfo::VersionInfo(const cxx::Serialization& serial) noexcept
+{
+    cxx::string<100> tmp_m_buildDateString;
+    cxx::string<100> tmp_commitIdString;
+    serial.extract(
+        m_versionMajor, m_versionMinor, m_versionPatch, m_versionTweak, tmp_m_buildDateString, tmp_commitIdString);
+    m_buildDateString = BuildDateStringType(cxx::TruncateToCapacity, tmp_m_buildDateString.c_str());
+    m_commitIdString = CommitIdStringType(cxx::TruncateToCapacity, tmp_commitIdString.c_str());
+}
+
+/// @brief Serialization of the VersionInfo.
+VersionInfo::operator cxx::Serialization() const noexcept
+{
+    cxx::string<100> tmp_m_buildDateString(cxx::TruncateToCapacity, m_buildDateString.c_str());
+    cxx::string<100> tmp_commitIdString(cxx::TruncateToCapacity, m_commitIdString.c_str());
+    return cxx::Serialization::create(
+        m_versionMajor, m_versionMinor, m_versionPatch, m_versionTweak, tmp_m_buildDateString, tmp_commitIdString);
+}
+
+bool VersionInfo::operator==(const VersionInfo& rhs) const noexcept
+{
+    return (m_versionMajor == rhs.m_versionMajor) && (m_versionMinor == rhs.m_versionMinor)
+           && (m_versionPatch == rhs.m_versionPatch) && (m_versionTweak == rhs.m_versionTweak)
+           && (m_buildDateString == rhs.m_buildDateString) && (m_commitIdString == rhs.m_commitIdString);
+}
+
+bool VersionInfo::operator!=(const VersionInfo& rhs) const noexcept
+{
+    return !(*this == rhs);
+}
+
+bool VersionInfo::checkCompatibility(const VersionInfo& other, CompatibilityCheckLevel compatibilityCheckLevel) const
+    noexcept
+{
+    bool isCompatible = false;
+    switch (compatibilityCheckLevel)
+    {
+    case CompatibilityCheckLevel::OFF:
+        isCompatible = true;
+        break;
+    case CompatibilityCheckLevel::MAJOR:
+        isCompatible = (m_versionMajor == other.m_versionMajor);
+        break;
+    case CompatibilityCheckLevel::MINOR:
+        isCompatible = (m_versionMajor == other.m_versionMajor) && (m_versionMinor == other.m_versionMinor);
+        break;
+    case CompatibilityCheckLevel::PATCH:
+        isCompatible = (m_versionMajor == other.m_versionMajor) && (m_versionMinor == other.m_versionMinor)
+                       && (m_versionPatch == other.m_versionPatch);
+        break;
+    case CompatibilityCheckLevel::COMMIT_ID:
+        isCompatible = (m_versionMajor == other.m_versionMajor) && (m_versionMinor == other.m_versionMinor)
+                       && (m_versionPatch == other.m_versionPatch) && (m_versionTweak == other.m_versionTweak)
+                       && (m_commitIdString == other.m_commitIdString);
+        break;
+    case CompatibilityCheckLevel::BUILD_DATE:
+        isCompatible = (m_versionMajor == other.m_versionMajor) && (m_versionMinor == other.m_versionMinor)
```

Comment:
nope, `(*this == other)` just compares two version infos. We need to compare unequal versions and check if they are compatible regarding the compatibility level, e.g. v1.2.3 would be compatible with v1.3.1 when the compatibility level is `MAJOR`, but incompatible if it's `MINOR`

Reply:
@elBoberido I think you misunderstood my comment here. I would use `isCompatible = (*this == other)` only for `CompatibilityCheckLevel::BUILD_DATE` since it is identical to the comparision operator. And in this specific case we can reduce the code doubling. 
Or maybe the comparision operator does use checkCompatibility with `BUILD_DATE`